### DISCUSSION TITLE
What is the correct language for Arduino Sketches?

### DISCUSSION THREAD

**Discussion by njbigwig** [community/community#56169] [Programming Help]
I created a repro with a single INO Sketch source file. When I look at languages:
repo:njbigwig/UnoGit  language:C++

Is there a way to change the language to Arduino or something more relevant? C++ does not seem to be the best choice.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @njbigwig ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by lennis-dev** [comment]
Hello @njbigwig 

I think you can find what you are looking for at the following [link](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-repository-languages)  of the GitHub documentation, also the Arduino programming language is C++/C, hope I could help.

Best regards
Lennis

**Comment by advayc** [comment]
arduino programming is a variant of C++ programming language. The code is written in C++ with an addition of special methods and functions.

**Comment by hamza-mis** [comment]
c++
